kymatio/scattering1d/frontend/base_frontend.py

```diff
@@ -594,99 +593,155 @@ def scattering(self, x):
-            # Sort paths by order and then by key 'n' (tuple)
-            # Order 0: n=(). Order 1: n=(n1, n_fr). Order 2: n=(n1, n2, n_fr)
-            S.sort(key=(lambda path: (path['order'], path['n'])))
-
-        if (self.out_type == 'dict'):
+        elif self.out_type == 'dict':
             return {path['n']: path['coef'] for path in S}
-        elif (self.out_type == 'list'):
+        elif self.out_type == 'list':
             return S
 
+    def meta(self):
+        backend = self._DryBackend()
+        filters = [self.phi_f, self.psi1_f, self.psi2_f]
+        U_gen = joint_timefrequency_scattering(None, backend,
+            filters, self.oversampling, self.average=='local',
+            self.filters_fr, self.oversampling_fr, self.average_fr=='local')
+        S_gen = jtfs_average_and_format(U_gen, backend,
+            self.phi_f, self.oversampling, self.average,
+            self.filters_fr[0], self.oversampling_fr, self.average_fr,
+            self.out_type, self.format)
+        S = sorted(list(S_gen), key=lambda path: (len(path['n']), path['n']))
+        meta = dict(order=np.array([len(path['n']) for path in S]))
+        meta['key'] = [path['n'] for path in S]
+        meta['n'] = []
+        meta['n_fr'] = []
+        for path in S:
+            if len(path['n']) == 0:
+                # Zeroth order: no n1, no n_fr, no n2
+                meta['n'].append([np.nan, np.nan])
+                meta['n_fr'].append(np.nan)
+            else:
+                n1_range = range(0, path['n1_max'], path['n1_stride'])
+                if len(path['n']) == 1:
+                    # First order and format='joint': n=(n_fr,)
+                    meta['n'].append([n1_range, np.nan])
+                elif len(path['n']) == 2 and self.format == 'joint':
+                    # Second order and format='joint': n=(n2, n_fr)
+                    meta['n'].append([n1_range, path['n'][0]])
+                elif len(path['n']) == 2 and self.format == 'time':
+                    # First order and format='time': n=(n1, n_fr)
+                    meta['n'].append([path['n'][0], np.nan])
+                elif len(path['n']) == 3 and self.format == 'time':
+                    # Second order and format='time': n=(n1, n2, n_fr)
+                    meta['n'].append([path['n'][2:]])
+                meta['n_fr'].append(path['n_fr'][0])
+        meta['n_fr'] = np.array(meta['n_fr'])
+        for key in ['xi', 'sigma', 'j']:
+            meta[key] = np.zeros((meta['n_fr'].shape[0], 2)) * np.nan
+            for order, filterbank in enumerate(filters[1:]):
+                for n, psi in enumerate(filterbank):
+                    meta[key][meta['n'][:, order]==n, order] = psi[key]
+            meta[key + '_fr'] = meta['n_fr'] * np.nan
+            for n_fr, psi_fr in enumerate(self.filters_fr[1]):
+                meta[key + '_fr'][meta['n_fr']==n_fr] = psi_fr[key]
+        meta['spin'] = np.sign(meta['xi_fr'])
+        return meta
+
+    def meta(self):
```

Comment:
why duplicate meta?

Reply:
I've added a commit that removes the duplicate definition of `meta`

Reply:
oops! this is definitely a rebase issue. thanks